### PR TITLE
ML framework pr

### PR DESCRIPTION
**Pull request by Olenek**
Framework with most of the features implemented, except for DNN and couple TODOs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,30 @@ If you want to collect only one type of cases, just use `--args="-p=a.txt"` for
 The procedure is identical with one key difference, use command `gradlew.bat` instead of `./gradlew`
 
 
-## Training the model
-
-The `RFTrain` directory contains Python scripts for evaluating different machine learning models on the collected dataset. The requirements are `python >=3.6`, `skipy`, `numpy`, and `pandas`.
+## Machine Learning
+
+Currently, there are a number of ML algorithms proposed as a valid models:
```
Comment (areyde) [inline README.md:28]: No `a` before `models`.

File: README.md [hunk 2]
```diff
@@ -23,6 +23,30 @@ If you want to collect only one type of cases, just use `--args="-p=a.txt"` for
...
+| GaussianNaiveBayes  | `sklearn.naive_bayes.GaussianNB`   | GNB |
+| ComplementNaiveBayes | `sklearn.naive_bayes.ComplementNB`   |CNB  |
+| DeepNeuralNetwork  | neural network built with `keras` | DNN |
+
+The `RFTrain` directory contains Python scripts for evaluating machine learning models on the collected datasets. 
+The requirements are specified in `requirements.txt`.
+
+### Training
+
+To train a model, one is ought to complete a list of procedures:
```
Comment (areyde) [inline README.md:46]: _One has to complete..._

File: RFTrain/ModelsTest.py [hunk 3]
```diff
@@ -208,17 +217,19 @@ def test_model_real(model):
...
-    #test_gauss()
-    #test_SVM()
-    #test_RF()
-    #test_NN()
+    # test_bernoulli()
+    # test_gauss()
+    # test_SVM()
+    # test_RF()
+    # test_NN()
     test_RF_CV()
```
Comment (areyde) [inline RFTrain/ModelsTest.py:234]: Is this just testing? Why are other models commented out? :)

File: RFTrain/src/model.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        return joblib.dump(self._model, os.path.join(self.model_train_path, 'trained.sav'))
+
+
+class OCCModel(Model):
+    def fit(self, neutrals):
+        return self._model.fit(neutrals)
+
+    def train(self, neutral, positive):
+        self.fit(neutral)
+        pos_accuracy = sum(self.predict(positive) == -1) / len(positive)
```
Comment (areyde) [inline RFTrain/src/model.py:37]: Can't this be done automatically?

File: RFTrain/src/model.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        with open(os.path.join(self.model_train_path, "training_results.txt"), 'a') as f:
+            to_print = "Training error fraction: {}\n" \
+                       "Accuracy on positives: {}\n".format(self._model.nu, pos_accuracy)
```
Comment (areyde) [inline RFTrain/src/model.py:40]: I believe writing as `f"{pos_accuracy}..." will be much simpler.

Comment (areyde) [reply RFTrain/src/model.py:40]: And everywhere below.

File: RFTrain/src/preprocessor.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        negatives = negatives.assign(label=lambda x: 0)  # Set label to zero, meaning negative
+        negatives = negatives.drop(columns=['score'])
+
+        whole_df = pd.concat([self.df_pos, negatives])
+        X = whole_df.drop(columns=['label'])
+        y = whole_df.label
+
+        return X, y
+
+    def make_neutral_positive(self, consider_quantile):
```
Comment (areyde) [inline RFTrain/src/preprocessor.py:28]: I would like some comments or doc-string here, to understand it.

File: RFTrain/src/utils.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+              'random_state': content['datasets'].get('random_seed'),
+              'model_type': content['model'].get('model_type'),
+              'model_config_path': os.path.join(content['model'].get('model_config_dir'),
+                                                content['model'].get('model_config_name')),
+              'model_train_dir': content['model'].get('model_train_dir')
+              }
+    return config
+
+
+def import_test_configuration(config_file):
```
Comment (areyde) [inline RFTrain/src/utils.py:24]: hmm

File: RFTrain/trainer.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        neutral, positive = dataset_preprocessor.make_neutral_positive(False)
+        model = OCCModel(config)
+        model.train(neutral, positive)
+
+    else:
+        features, targets = dataset_preprocessor.make_binary()
+        model = BinaryModel(config)
+        model.train(features, targets)
+        # TODO: cross validate-split to config
+        # TODO: save model info etc.
```
Comment (areyde) [inline RFTrain/trainer.py:21]: todo

**Review by areyde — CHANGES_REQUESTED** [review]
💯 Some small fixes here and there!

File: RFTrain/src/model.py (hunk 4 above)
Comment (Olenek) [reply RFTrain/src/model.py:40]: I'm not sure. sklearn wiki does not provide examples on how to estimate accuracy of one class models

File: RFTrain/src/model.py (hunk 5 above)
Comment (Olenek) [reply RFTrain/src/model.py]: changed to your preference

File: RFTrain/src/preprocessor.py (hunk 6 above)
Comment (Olenek) [reply RFTrain/src/preprocessor.py]: done
